lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -299,15 +364,37 @@ def test_user_groups(self):
         self.assertEqual(views.user_groups(mock_user), [])
 
     def test_get_current_child(self):
-        self.assertIsNone(views.get_current_child(MagicMock()))
         mock_xmodule = MagicMock()
+        self.assertIsNone(views.get_current_child(mock_xmodule))
+
         mock_xmodule.position = -1
-        mock_xmodule.get_display_items.return_value = ['one', 'two']
+        mock_xmodule.get_display_items.return_value = ['one', 'two', 'three']
         self.assertEqual(views.get_current_child(mock_xmodule), 'one')
-        mock_xmodule_2 = MagicMock()
-        mock_xmodule_2.position = 3
-        mock_xmodule_2.get_display_items.return_value = []
-        self.assertIsNone(views.get_current_child(mock_xmodule_2))
+
+        mock_xmodule.position = 2
+        self.assertEqual(views.get_current_child(mock_xmodule), 'two')
+        self.assertEqual(views.get_current_child(mock_xmodule, requested_child='first'), 'one')
+        self.assertEqual(views.get_current_child(mock_xmodule, requested_child='last'), 'three')
+
+        mock_xmodule.position = 3
+        mock_xmodule.get_display_items.return_value = []
+        self.assertIsNone(views.get_current_child(mock_xmodule))
+
+    def test_get_redirect_url(self):
+        self.assertIn(
+            'activate_block_id',
+            get_redirect_url(self.course_key, self.section.location),
+        )
+
+        self.assertIn(
+            '&child=first',
+            get_redirect_url(self.course_key, self.section.location, child='first'),
+        )
+
+        self.assertIn(
+            '&child=last',
+            get_redirect_url(self.course_key, self.section.location, child='last'),
+        )
```

Comment:
More "stringly typed" fragility. Can you parse the URL, and test against a dict of the actual params?  

The current version could break if:

1. The `child` param came first (and was therefore represented as `?child=last` rather than `&child=last`
2. The param value did not end at the end of the given string (`&child=lastory`).  Maybe wrap in in a `self.assert_url_has_params` method?